### PR TITLE
Implement `geom_parallel_slopes()`

### LINKED ISSUE #26: Plotting parallel slopes model
When considering multiple regression with one numerical and one categorical predictor, the `ggplot` code to plot the interaction model is easy while the code to plot the "parallel slopes" model is too hard. Consider adding a `geom_parallel_slopes()` ggplot2 geometry extension to `moderndive` package that behaves like `geom_smooth(method = "lm", ...)`

## Example: Seattle house prices

``` r
suppressPackageStartupMessages(library(tidyverse))
library(knitr)
library(moderndive)

# log10() transformations
house_prices <- house_prices %>%
  mutate(
    log10_price = log10(price),
    log10_size = log10(sqft_living)
  )
```

## Default plot is based on interaction model

Default plot when using col = condition is interaction model: different slopes and intercepts

``` r
ggplot(house_prices, aes(x = log10_size, y = log10_price, col = condition)) +
  geom_point(alpha = 0.1) +
  geom_smooth(method = "lm", se = FALSE, size = 1) +
  labs(x = "log10 square feet living space", y = "log10 price in USD", title = "House prices in Seattle: Interaction model")
```

Corresponding slopes and intercepts
``` r
lm(log10_price ~ log10_size * condition, data = house_prices) %>% 
  get_regression_table()
#> # A tibble: 10 x 7
#>    term             estimate std_error statistic p_value lower_ci upper_ci
#>    <chr>               <dbl>     <dbl>     <dbl>   <dbl>    <dbl>    <dbl>
#>  1 intercept           3.33      0.451     7.38    0        2.45     4.22 
#>  2 log10_size          0.69      0.148     4.65    0        0.399    0.98 
#>  3 condition2          0.047     0.498     0.094   0.925   -0.93     1.02 
#>  4 condition3         -0.367     0.452    -0.812   0.417   -1.25     0.519
#>  5 condition4         -0.398     0.453    -0.879   0.38    -1.29     0.49 
#>  6 condition5         -0.883     0.457    -1.93    0.053   -1.78     0.013
#>  7 log10_size:cond…   -0.024     0.163    -0.148   0.882   -0.344    0.295
#>  8 log10_size:cond…    0.133     0.148     0.893   0.372   -0.158    0.424
#>  9 log10_size:cond…    0.146     0.149     0.979   0.328   -0.146    0.437
#> 10 log10_size:cond…    0.31      0.15      2.07    0.039    0.016    0.604
```

## Parallel slopes is too hard

Compute all regression points first

``` r
model_price_3_points <-
  house_prices %>%
  lm(log10_price ~ log10_size + condition, data = .) %>%
  get_regression_points()
```
Display a random sample of these regression points
```r
model_price_3_points %>% 
  group_by(condition) %>% 
  sample_n(2)
#> # A tibble: 10 x 6
#> # Groups:   condition [5]
#>       ID log10_price log10_size condition log10_price_hat residual
#>    <int>       <dbl>      <dbl> <fct>               <dbl>    <dbl>
#>  1  7377        5.47       2.86 1                    5.27    0.196
#>  2 15372        5.82       3.00 1                    5.40    0.421
#>  3  9277        5.52       3.00 2                    5.35    0.166
#>  4  5859        5.34       2.96 2                    5.32    0.022
#>  5 19748        5.80       3.38 3                    5.74    0.057
#>  6 13959        5.74       3.28 3                    5.66    0.077
#>  7 17844        5.48       3.02 4                    5.45    0.024
#>  8  5380        5.89       3.36 4                    5.74    0.148
#>  9 18642        5.63       3.54 5                    5.94   -0.305
#> 10  7422        5.69       3.01 5                    5.50    0.189
```

Very cumbersome code to plot
```r
ggplot(house_prices, aes(x = log10_size, y = log10_price, col = condition)) +
  geom_point(alpha = 0.1) +
  labs(y = "log10 price", x = "log10 square footage", title = "House prices in Seattle: Parallel slopes model") +
  geom_line(data = model_price_3_points, aes(y = log10_price_hat), size = 1.5, show.legend = FALSE) +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))
```

Corresponding slope and intercepts
``` r
lm(log10_price ~ log10_size + condition, data = house_prices) %>% 
  get_regression_table()
#> # A tibble: 6 x 7
#>   term       estimate std_error statistic p_value lower_ci upper_ci
#>   <chr>         <dbl>     <dbl>     <dbl>   <dbl>    <dbl>    <dbl>
#> 1 intercept     2.88      0.036     80.0    0        2.81     2.95 
#> 2 log10_size    0.837     0.006    134.     0        0.825    0.85 
#> 3 condition2   -0.039     0.033     -1.16   0.246   -0.104    0.027
#> 4 condition3    0.032     0.031      1.04   0.3     -0.028    0.092
#> 5 condition4    0.044     0.031      1.42   0.155   -0.017    0.104
#> 6 condition5    0.096     0.031      3.09   0.002    0.035    0.156
```

Created on 2018-07-08 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0.9000).

### PR DESCRIPTION
**Pull request by echasnovski**
This PR implements a {ggplot2} layer function for fitting and plotting parallel slopes model. It is as similar to `geom_smooth()` as reasonably possible. It also has the ability to fit non-linear models (as shown in examples).

Closes #26.

### CODE REVIEW

**Comment by echasnovski** [comment]
I updated '.travis.yml' to not run {vdiffr} tests on old release. They are both redundant and currently have issues with different random generation algorithm. This is currently a problem because I set up tests for `geom_parallel_slopes()` with some random generation. If you want to preserve {vdiffr} tests on R old release, tell me and I'll redo tests with different (determenistic) example data set.

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-r:
-  - oldrel
-  - release
-  - devel
+
+matrix:
+  include:
+  - r: oldrel
+    # Don't use vdiffr on old R release
+    env: VDIFFR_RUN_TESTS=false
```
Comment (rudeboybert) [inline .travis.yml]: interesting

File: R/geom_parallel_slopes.R [hunk 2]
```diff
@@ -0,0 +1,168 @@
+#' Parallel slopes model
+#' 
+#' \code{geom_parallel_slopes()} fits parallel slopes model and adds its line
+#' output(s) to a \code{ggplot} object. Basically, it fits a unified model with
+#' intercepts varying between groups (which should be supplied as standard
+#' {ggplot2} grouping aesthetics: \code{group}, \code{color}, \code{fill},
+#' etc.).
```
Comment (rudeboybert) [inline R/geom_parallel_slopes.R]: Could you add a comment that this geom provides functionality that `geom_smooth()` doesn't?

File: R/geom_parallel_slopes.R [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+#' example_df <- house_prices %>%
+#'   dplyr::slice(1:1000) %>% 
+#'   dplyr::mutate(
+#'     log10_price = log10(price),
+#'     log10_size = log10(sqft_living)
+#'   )
+#' ggplot_viz <- ggplot2::ggplot(
+#'   example_df,
+#'   ggplot2::aes(x = log10_size, y = log10_price, color = condition)
+#' ) +
```
Comment (rudeboybert) [inline R/geom_parallel_slopes.R]: After we merge we'll likely polish these a bit to make them more in line with the coding style we've been presenting to students in our intro courses. For example, we don't use `dplyr::` but rather load the libraries

**Comment by rudeboybert** [comment]
Thanks a bunch! This really great work! Also, I just learned a lot about unit testing in this PR!

**Comment by echasnovski** [comment]
I am really glad this PR was and will be useful. Thank you for a challenging task :+1:

**Comment by rudeboybert** [comment]
Thank you for doing it on such short notice! Everyone I've shown this to is already excited!

**Pull request merged**